Bugsnag/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -143,8 +143,7 @@ void bsg_kscrw_i_addIntegerElement(const BSG_KSCrashReportWriter *const writer,
 void bsg_kscrw_i_addUIntegerElement(const BSG_KSCrashReportWriter *const writer,
                                     const char *const key,
                                     const unsigned long long value) {
-    bsg_ksjsonaddIntegerElement(bsg_getJsonContext(writer), key,
-                                (long long)value);
+    bsg_ksjsonaddUIntegerElement(bsg_getJsonContext(writer), key, value);
```

Comment:
This was originally cast to signed because JSON technically doesn't support 64-bit unsigned values. But this was written before I learned that technically JSON doesn't even support 63-bit signed values (it's actually 54 bits, the coefficient size in 64-bit ieee754 floats) :P.

Regardless, we need to be sure that upstream processors (with each language's different JSON implementations) are able to handle this change. I believe that some of them always assume signed ints.

Reply:
Based on my reading of [the spec](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf), I don't think JSON imposes any size limits on what may be stored.

> JSON is agnostic about the semantics of numbers. In any programming language, there can be a variety of number types of various capacities and complements, fixed or floating, binary or decimal. That can make interchange between different programming languages difficult. JSON instead offers only the representation of numbers that humans use: a sequence of digits. All programming languages know how to make sense of digit sequences even if they disagree on internal representations. That is enough to allow interchange.

That said some parsers will inevitably get this wrong and impose their own limits.

If it is the case that our upstream processors don't support 64-bit unsigned values, then it doesn't make sense to incorporate these changes from KSCrash.

Reply:
I'd actually be fine with the changes, so long as we can verify that our pipeline can handle it upstream.

Reply:
Also, we need to check if our upstream code expects signed ints and tries to convert to unsigned, considering that's how the source JSON was formatted until now...

Reply:
Opened PLAT-5084 to cover the pipeline / dashboard handling